src/management-system-v2/components/properties-panel.tsx

```diff
@@ -2,206 +2,364 @@
 
 import type Modeling from 'bpmn-js/lib/features/modeling/Modeling';
 
+import { getFillColor, getStrokeColor } from 'bpmn-js/lib/draw/BpmnRenderUtil';
+
 import type { ElementLike } from 'diagram-js/lib/core/Types';
 
 import useModelerStateStore from '@/lib/use-modeler-state-store';
 
-import React, { FocusEvent, useEffect, useState } from 'react';
+import React, { FocusEvent, useCallback, useEffect, useMemo, useState } from 'react';
 
-import { Card, Input, ColorPicker, Drawer, Space, Image } from 'antd';
+import { Input, ColorPicker, Space, Image } from 'antd';
 
 import { EuroCircleOutlined, ClockCircleOutlined } from '@ant-design/icons';
+import BpmnFactory from 'bpmn-js/lib/features/modeling/BpmnFactory';
+import {
+  getMetaDataFromElement,
+  getMilestonesFromElement,
+  setProceedElement,
+} from '@proceed/bpmn-helper';
+import CustomPropertySection from './custom-property-section';
+import MilestoneSelectionSection from './milestone-selection-section';
+import ResizableElement from './ResizableElement';
+import CollapsibleCard from './collapsible-card';
 
 type PropertiesPanelProperties = {
   selectedElement: ElementLike;
   setOpen: (open: boolean) => void;
 };
 
 const PropertiesPanel: React.FC<PropertiesPanelProperties> = ({ selectedElement, setOpen }) => {
+  const [showInfo, setShowInfo] = useState(true);
+  const [width, setWidth] = useState(450);
+
+  const [metaData, setMetaData] = useState(getMetaDataFromElement(selectedElement.businessObject));
+  const [milestones, setMilestones] = useState(
+    getMilestonesFromElement(selectedElement.businessObject),
+  );
   const [name, setName] = useState('');
```

Comment:
This wasn't part of your PR, but derived state shouldn't be stored as additional state, could you change this to a normal variable?